src/procedures/proc_functions.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+
+
+#include "../util/arr.h"
+#include "rax.h"
+#include "../RG.h"
+#include "procedure.h"
+#include "proc_functions.h"
+#include "../datatypes/array.h"
+
+extern rax *__aeRegisteredFuncs;
+
+// CALL dbms.functions()
+
+typedef struct {
+    SIValue *output;                      // array with a one entry: [name]
+    raxIterator iter;                     // procedures iterator
+    SIValue *yield_name;                  // yield name
+    SIValue *yield_signature;             // yield the function signature in the format name: t1 ... tn -> tn+1
+    SIValue *yield_description;           // yield document string
+    SIValue *yield_return_description;    // yield the description of the return value of the function
+    SIValue *yield_argument_description;  // yield the description of the function arguments
+} ProcFunctionsPrivateData;
+
+static void _process_yield
```

Comment:
Please make sure each function is documented.
docs should include a short description explaining what the function do, in addition to each argument